executable/swf.ksy

```diff
@@ -104,7 +152,184 @@ types:
+            'tag_type::define_sprite': define_sprite_body
             'tag_type::export_assets': symbol_class_body
+            'tag_type::define_shape4': define_shape4_body
+  define_shape4_body:
+    seq:
+      - id: id
+        -orig-id: ShapeId
+        type: u2
+      - id: shape_bounds
+        type: rect
+      - id: edge_bounds
+        type: rect
+      - id: reserved
+        type: b5
+      - id: use_fill_winding_rule
+        type: b1
+      - id: use_non_scaling_stroke
+        type: b1
+      - id: use_scaling_stroke
+        type: b1
+      - id: shapes
+        # We need to carry the version of the shape as it impact the layout of the struct down the tree
+        type: shape_with_style(4)
+  shape_with_style:
+    params:
+      - id: shape_version
+        type: u1
+    seq:
+      - id: fill_styles
+        type: fill_style_array(shape_version)
+      - id: line_styles
+        type: line_style_array(shape_version)
+  line_style_array:
+    params:
+      - id: shape_version
+        type: u1
+    seq:
+      - id: line_style_count
+        type: u1
+      - id: line_style_count_extended
+        type: u2
+        if: line_style_count == 0xFF
+      - id: line_styles
+        type: line_style(shape_version)
+        repeat: expr
+        repeat-expr: line_style_count
+        if: shape_version != 4
+      - id: line_styles2
+        type: line_style2(shape_version)
+        repeat: expr
+        repeat-expr: line_style_count
+        if: shape_version == 4
+  line_style:
+    params:
+      - id: shape_version
+        type: u1
+    seq:
+      - id: width
+        type: u2
+      - id: color_a
+        type: rgba
+        if: shape_version == 4
+      - id: color
+        type: rgb
+        if: shape_version != 4
+  line_style2:
+    params:
+      - id: shape_version
+        type: u1
+    seq:
+      - id: width
+        type: u2
+      - id: start_cap_style
+        type: b2
+        enum: cap_type
+      - id: join_style
+        type: b2
+        enum: join_style
+      - id: has_fill_flag
+        type: b1
+      - id: no_h_scale_flag
+        type: b1
+      - id: no_v_scale_flag
+        type: b1
+      - id: pixel_hinting_flag
+        type: b1
+      - id: reserved
+        type: b5
+      - id: no_close
+        type: b1
+      - id: end_cap_style
+        type: b2
+        enum: cap_type
+      - id: meter_limit_factor
+        type: u2
+        if: join_style == join_style::miter_join
+      - id: color
+        type: rgba
+        if: has_fill_flag == false
+      - id: fill_type
+        type: fill_style(shape_version)
+        if: has_fill_flag == true
+  fill_style_array:
+    params:
+      - id: shape_version
+        type: u1
+    seq:
+      - id: fill_style_count
+        type: u1
+      - id: fill_style_count_extended
+        type: u2
+        if: fill_style_count == 0xFF
+      - id: fill_styles
+        type: fill_style(shape_version)
+        repeat: expr
+        repeat-expr: fill_style_count
```

Comment:
Is it really a `fill_style_count` but not `(fill_style_count!=0xff?fill_style_count:fill_style_count_extended)`?

Reply:
You are right, I had no idea kaitai supporterd if / ternary operator in the repeat-expr field. Will fix (still new to kaitai here 😅 )

Reply:
Fixed in 1efe4b15c689ea88561c6191116d47a976736882